### PR TITLE
Add colorblind-safe visualization theme and resolve merge conflict

### LINKED ISSUE #123: [FEATURE]: Add native backend resolver for Hugging Face transformers
### Summary

**Problem Statement**
Currently, TrustLens offers a seamless, zero-friction auditing experience for traditional ML models via native backend resolvers (`sklearn.py`, `xgboost.py`). However, as practitioners increasingly rely on fine-tuned LLMs and BERT variants for text classification, they are blocked. To audit a Hugging Face model today, a user must manually extract raw logits, apply softmax, align string labels, and convert everything to NumPy arrays before calling `analyze()`. This introduces massive friction for the NLP community.

**Proposed Solution**
We need to introduce a native backend resolver for the Hugging Face ecosystem. 
Because TrustLens uses a decoupled `PredictionBundle` architecture, we can easily plug this in without changing the core analysis engine.

### Where to Edit
1. **`trustlens/backends/huggingface.py`** (New File): Implement the resolver logic.
2. **`trustlens/backends/registry.py`**: Register the new framework.
3. **`tests/test_huggingface_backend.py`** (New File): Add basic unit tests.
4. **`pyproject.toml`**: Add optional `nlp` dependency group.

### Proposed Implementation Steps

**1. Create the Resolver (`trustlens/backends/huggingface.py`)**
Create a `resolve()` function that matches the signature in `sklearn.py`. 
* Check if the `model` is a `transformers.Pipeline` (specifically `TextClassificationPipeline`).
* If `y_prob` is not provided, pass the input strings `X` through the pipeline: `model(X, top_k=None)`.
* Parse the returned list of dictionaries (which contain `{'label': ..., 'score': ...}`).
* Construct a 2D `np.ndarray` of probabilities (`y_prob`).
* Construct a 1D `np.ndarray` of predicted integer indices (`y_pred`).
* Return a `PredictionBundle(y_pred=y_pred, y_prob=y_prob, framework="huggingface", class_labels=...)`.

**2. Update the Registry (`trustlens/backends/registry.py`)**
* Add `"transformers": "huggingface"` to the `FRAMEWORK_MAPPING` dictionary.
* Update `IMPLEMENTED_RESOLVERS` to include `"huggingface"`.
* In `get_resolver()`, add the routing logic:
  ```python
  if detected == "huggingface":
      from trustlens.backends import huggingface
      return huggingface.resolve
  ```

**3. Optional Dependencies (`pyproject.toml`)**
Because the `transformers` and `torch` libraries are massive, this integration **must** be implemented as an optional dependency to avoid bloating the core package for tabular users. 
Add an optional dependency group:
```toml
[project.optional-dependencies]
nlp = ["transformers", "torch"]
```
*(Ensure all imports inside `huggingface.py` are wrapped in `try/except ImportError` blocks).*

**Alternatives Considered**
We could maintain the status quo and require NLP users to pass `framework="manual"` alongside pre-computed `y_prob` arrays. However, because Hugging Face is the undisputed industry standard for NLP, missing native support limits TrustLens's adoption. 

### How to Verify
Write a mock test in `tests/test_huggingface_backend.py`. You do not need to download a massive model; you can use the tiny `hf-internal-testing/tiny-random-DistilBertForSequenceClassification` or simply mock the `Pipeline` object to ensure it returns the correct NumPy arrays.

```bash
pytest tests/test_huggingface_backend.py
```

### 🤝 Need Help?
This is a fantastic issue for a contributor looking to dive into the core architecture of TrustLens while working with modern NLP tools! If you have any questions about the `PredictionBundle` contract or Hugging Face pipelines, feel free to comment on this issue or join our Discussions.

### PR DESCRIPTION
**Pull request by Software566**
# Pull Request Template

## Type of Change
Please check the type of change:
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 📝 Documentation update (changes to docs or examples)
- [ ] 🧹 Maintenance (refactoring, code cleanup, dependencies)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Changes Made
- [ ] Describe change 1
- [ ] Describe change 2

## Testing

- [ ] All unit tests pass locally.
- [ ] Added new tests for the changes made.

## Pre-submit Checklist
Before submitting, please ensure you have:
- [ ] Run `pre-commit run --all-files` and all hooks passed.
- [ ] Verified that all tests pass (`make test`).
- [ ] Updated the `CHANGELOG.md` if applicable.
- [ ] Verified that documentation has been updated for new features.

## Summary by CodeRabbit

* **New Features**
  * Added a colorblind-friendly visualization theme with improved color accessibility for categorical data and severity indicators.

### CODE REVIEW

**Review by Khanz9664 — CHANGES_REQUESTED** [review]
@Software566 Thanks for the contribution and for working on accessibility improvements.

I reviewed the implementation and found one blocking issue before this can be merged.

The new `COLORBLIND_THEME` replaces the entire semantic mapping with `COLORBLIND_SEMANTIC_COLORS`, but that mapping currently only defines `"severity"`. Existing visualizations expect additional semantic groups such as `"verdict"`, `"grade"`, `"direction"`, and `"neutral"`. As a result, using `COLORBLIND_THEME` can raise `KeyError` at runtime when those mappings are accessed.

I recommend constructing the colorblind theme by starting from the existing `SEMANTIC_COLORS` mapping and overriding only the entries that need colorblind-safe replacements. That preserves compatibility with the rest of the visualization layer while still providing the new palette.

Additionally, it looks like several detailed docstrings added during the recent documentation initiative were reduced to single-line descriptions. Since those expanded docstrings were intentionally added to improve contributor onboarding and API documentation quality, please preserve the existing documentation sections when updating this module.

Once the semantic mapping issue is resolved and the documentation regression is addressed, I'd be happy to take another look.
